src/formatDate.js

```diff
@@ -8,7 +8,24 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = Array.from(fromFormat).pop();
+  const newFormat = Array.from(toFormat);
+  const newSeparator = newFormat.pop();
+  const dataObject = {};
+
+  for (const value of date.split(oldSeparator)) {
+    const key = fromFormat.shift();
+
+    dataObject[key] = value;
+
+    if (key === 'YY') {
+      dataObject.YYYY = value < 30 ? '20' + value : '19' + value;
```

Comment:
When converting from 'YY' to 'YYYY', the assumption that any 'YY' less than '30' belongs to the 21st century may not always hold true. This could lead to incorrect dates for cases that don't follow this rule. If this rule is part of the function requirements, it should be clearly documented.